src/app/widgets/exchange/header/header.component.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  Exchange,
+  ExchangeResult,
+  Rate,
+  Rates,
+} from "./../interfaces/Currency";
+import { Component, OnInit } from "@angular/core";
+import { SelectedCurrencyIdService } from "../services/selected-currency-id.service";
+import { CurrenciesService } from "../services/currenciesList.service";
+
+@Component({
+  selector: "app-header",
+  templateUrl: "./header.component.html",
+  styleUrls: ["./header.component.scss"],
+})
+export class HeaderComponent implements OnInit {
+  exchange!: ExchangeResult;
+  rate!: Rates;
+
+  currentCurrency!: Exchange;
+  currencyRate: Rate | null = null;
+
+  constructor(
+    public selectedCurrencyService: SelectedCurrencyIdService,
+    private currencyRateService: CurrenciesService
+  ) {}
+
+  ngOnInit() {
+    this.selectedCurrencyService.selectedCurrency.subscribe((exchange) => {
+      this.currentCurrency = exchange;
+      const pairs = `${this.currentCurrency.currencyFrom}${this.currentCurrency.currencyTo}`;
+      this.currencyRateService.requestRates(pairs).subscribe((response) => {
+        this.currencyRate = response.rates[pairs];
+      });
+    });
+  }
+
+  todayDate: Date = new Date();
```

Comment:
This could be moved to the top, all variable declarations are good to be on top of our file, between class declaration and constructor.